src/com/sap/piper/integration/TransportManagementService.groovy

```diff
@@ -156,6 +131,144 @@ class TransportManagementService implements Serializable {
+        }
+
+        def proxy = config.proxy ? config.proxy : script.env.HTTP_PROXY
+
+        def responseExtDescriptorUpdate = 'responseExtDescriptorUpdate.txt'
+
+        def responseCode = script.sh returnStdout: true,
+                                     script: """|#!/bin/sh -e
+                                                | curl ${proxy ? '--proxy ' + proxy + ' ' : ''} \\
+                                                |      --write-out '%{response_code}' \\
+                                                |      -H 'Authorization: Bearer ${token}' \\
+                                                |      -H 'tms-named-user: ${namedUser}' \\
+                                                |      -F 'file=@${file}' \\
+                                                |      -F 'mtaVersion=${mtaVersion}' \\
+                                                |      -F 'description=${description}' \\
+                                                |      --output ${responseExtDescriptorUpdate} \\
+                                                |      -X PUT \\
+                                                |      '${url}/v2/nodes/${nodeId}/mtaExtDescriptors/${idOfMtaExtDescriptor}'""".stripMargin()
+
+        return jsonUtils.jsonStringToGroovyObject(getResponseBody(responseExtDescriptorUpdate, responseCode, '200', 'MTA Extension Descriptor Update'))
+    }
+
+    def getMtaExtDescriptor(String url, String token, Long nodeId, String mtaId, String mtaVersion) {
+        echo("Get Extension descriptor started.")
+
+        if (config.verbose) {
+            echo("URL: '${url}', NodeId: '${nodeId}', MtaId: '${mtaId}', MtaVersion: '${mtaVersion}'")
+        }
+
+        def parameters = [
+            url          : "${url}/v2/nodes/${nodeId}/mtaExtDescriptors?mtaId=${mtaId}&mtaVersion=${mtaVersion}",
+            httpMode     : "GET",
+            contentType  : 'APPLICATION_JSON',
+            customHeaders: [
+                [
+                    maskValue: true,
+                    name     : 'authorization',
+                    value    : "Bearer ${token}"
+                ]
+            ]
+        ]
+
+        def proxy = config.proxy ? config.proxy : script.env.HTTP_PROXY
+
+        if (proxy){
+            parameters["httpProxy"] = proxy
+        }
+
+        def response = sendApiRequest(parameters)
+        if (response.status != 200) {
+            prepareAndThrowException(response, "Get Extension descriptor failed (HTTP status code '${response.status}').")
+        }
+
+        if (config.verbose) {
+            echo("Response content '${response.content}'.")
+        }
+
+        // because the API is called with params, the return is always a map with either a empty list or a list containing one element
+        Map mtaExtDescriptor = [:]
+        Map responseContent = jsonUtils.jsonStringToGroovyObject(response.content)
+        if(responseContent.get("mtaExtDescriptors")) {
```

Comment:
> Should we throw an exception (call `error`) here in case case we can't get the extension descriptor from the response (... the parsed entity does not have a node mtaExeDescriptor). Currenty we continue with the empty map silently. Can / will that cause a follow-up error later? In that case it would be better to be fail fast here. In case it considered to be a normal situation that there is no node `mtaExtDescriptors` it is fine. I'm just asking and are not deeply familiarised with the use case.

Hi Marcus, thanks for mention this good point! Yes, a node without a mtaExtDescriptors is fine. As the code has checked the response status in Line 253. In the case of HTTP 200, the response body can only be a map with an empty list or a one-element list.

Reply:
ok